microsoft/testsuites/power/power.py

```diff
@@ -0,0 +1,587 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import re
+import time
+from datetime import date
+from typing import List
+
+from assertpy import assert_that
+
+from lisa import (
+    Environment,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    schema,
+)
+from lisa.base_tools import Cat, Uname
+from lisa.features import Disk, StartStop
+from lisa.operating_system import Redhat, Suse, Ubuntu
+from lisa.testsuite import TestCaseMetadata
+from lisa.tools import Blkid, Dmesg, Echo, Fdisk, Sed, Swap, SwapOn
+from lisa.util import LisaException, SkippedException, find_patterns_in_lines
+from lisa.util.logger import Logger
+from lisa.util.shell import wait_tcp_port_ready
+
+
+def check_vm_feature_support_status(min_support_version: str, node: Node) -> bool:
+    if not min_support_version:
+        raise SkippedException("No min support kernel version provided!")
+    
+    kernel_version = node.tools[Uname].get_linux_information().kernel_version
+    if kernel_version > min_support_version:
+        return True
+    elif kernel_version < min_support_version:
+        return False
+    
+    return True
+
+def get_year(node: Node) -> str:
+    hard_ware_clock = node.execute(
+        "hwclock",
+        shell=True,
+        sudo=True
+    ).stdout
+    if isinstance(node.os, Ubuntu) and node.os.information.version == '16.4.0':
+        year = re.split('\s+', hard_ware_clock)[3]
```

Comment:
1. provide example for each pattern
2. use re.compile like other examples.
3. use match to capture, instead of use array list, it's not easy to maintain.

Reply:
1. added
2. and 3: do you mean writing it this way:
`hwclock = '2022-02-04 11:38:06.949136+0000'`
`pattern = re.compile("^\d{4}")`
`year = re.match(pattern, hwclock)`
`print(year[0])`

Reply:
Similar. the `util/__init__.py` has some implementations of matching, it can be reused.